apollo-router/src/services/layers/persisted_queries/manifest_poller.rs

```diff
@@ -207,7 +207,76 @@ impl PersistedQueryManifestPoller {
     /// Starts polling immediately and this function only returns after all chunks have been fetched
     /// and the [`PersistedQueryManifest`] has been fully populated.
     pub(crate) async fn new(config: Configuration) -> Result<Self, BoxError> {
-        if let Some(uplink_config) = config.uplink.as_ref() {
+        if let Some(local_pq_list) = config.persisted_queries.local_manifest {
+            tracing::info!(
+                "Loading persisted query list from local file: {}",
+                local_pq_list.clone()
+            );
+            let local_manifest =
+                std::fs::read_to_string(local_pq_list.clone()).map_err(|e| -> BoxError {
```

Comment:
As we're in an async function I would feel more comfortable if we used this [async version](https://docs.rs/tokio_wasi/latest/tokio/fs/fn.read_to_string.html) instead

Reply:
Updated to use the async variant, although is using await in it; let me know if there's a different approach preferred here.